tool/src/c/results.rs

```diff
@@ -42,28 +41,24 @@ pub fn collect_results<'a>(
     }
 }
 
-pub fn gen_result<W: fmt::Write>(
-    typ: &ast::TypeName,
-    in_path: &ast::Path,
-    env: &Env,
-    out: &mut W,
-) -> fmt::Result {
+pub fn gen_result<W: fmt::Write>(typ: &ast::TypeName, env: &Env, out: &mut W) -> fmt::Result {
     if let ast::TypeName::Result(ok, err) = typ {
-        let result_name = format!("{}_{}", in_path.elements.join("_"), name_for_type(typ));
```

Comment:
So here we're removing the prefixes to make the paths simpler, right?

Reply:
Yes.

There's no actual namespacing needed for the result itself since two results with the same types are the same result